sg_jira/jira_session.py

```diff
@@ -91,7 +91,12 @@ def setup(self):
         """
         # Build a mapping from Jira field names to their id for fast lookup.
         for jira_field in self.fields():
-            self._jira_fields_map[jira_field["name"].lower()] = jira_field["id"]
+            field_name = jira_field["name"].lower()
+            if field_name == "due date":
```

Comment:
Tbh, I don't like having a specific condition for the "Due Date". For now, we only see this happening with this field but maybe there are other cases. Would it be possible to implement something more generic?
Maybe we could implement a fallback logic using other fields rather than the "name" field when comparing values?
What about the "key" or "clauseNames" returned by `jira.fields()`

```
>>> pprint.pprint(b.jira.fields())
[
  {'clauseNames': ['due', 'duedate'],
    'custom': False,
    'id': 'duedate',
    'key': 'duedate',
    'name': 'Due date',
    'navigable': True,
    'orderable': True,
    'schema': {'system': 'duedate', 'type': 'date'},
    'searchable': True},
]
```